internal/v1/handler.go

```diff
@@ -498,6 +498,89 @@ func convertIgnoreImageTypeToSlice(ignoreImageTypes *[]ImageTypes) []string {
 	return stringSlice
 }
 
+func newComposeResponse(data []ComposesResponseItem, count int, first, last string) ComposesResponse {
+	return ComposesResponse{
+		Data: data,
+		Meta: struct {
+			Count int `json:"count"`
+		}{
+			count,
+		},
+		Links: struct {
+			First string `json:"first"`
+			Last  string `json:"last"`
+		}{first, last},
+	}
+}
+
+func (h *Handlers) getComposesResponse(specVersion string, orgId string, since time.Duration, limit, offset int, ignoreImageTypeStrings []string) (ComposesResponse, error) {
+	composes, count, err := h.server.db.GetComposes(orgId, since, limit, offset, ignoreImageTypeStrings)
+	if err != nil {
+		return ComposesResponse{}, err
+	}
+
+	data := []ComposesResponseItem{}
+	for _, c := range composes {
+		var cmpr ComposeRequest
+		err = json.Unmarshal(c.Request, &cmpr)
+		if err != nil {
+			return ComposesResponse{}, err
+		}
+		data = append(data, ComposesResponseItem{
+			CreatedAt: c.CreatedAt.Format(time.RFC3339),
+			Id:        c.Id,
+			ImageName: c.ImageName,
+			Request:   cmpr,
+			ClientId:  (*ClientId)(c.ClientId),
+		})
+	}
+
+	lastOffset := count - 1
+	if lastOffset < 0 {
+		lastOffset = 0
+	}
+	first := fmt.Sprintf("%v/v%v/composes?offset=0&limit=%v", RoutePrefix(), specVersion, limit)
+	last := fmt.Sprintf("%v/v%v/composes?offset=%v&limit=%v", RoutePrefix(), specVersion, lastOffset, limit)
+	return newComposeResponse(data, count, first, last), nil
```

Comment:
I wonder if the string building should be part of the helper function, it is gonna be always the same.